src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -517,6 +517,23 @@ public T WithTelemetry(ITelemetryConfig telemetryConfig)
             return (T)this;
         }
 
+        /// <summary>
+        /// Sets telemetry client for the application.
+        /// </summary>
+        /// <param name="telemetryClients">List of telemetry clients to add telemetry logs.</param>
+        /// <returns>The builder to chain the .With methods</returns>
+        public T WithTelemetryClient(params ITelemetryClient[] telemetryClients)
+        {
+            ValidateUseOfExperimentalFeature("ITelemetryClient");
+
+            if ((telemetryClients != null) && telemetryClients.Any())
+            {
```

Comment:
Thoughts on throwing if `telemetryClients` is null?

Reply:
The SAL implementation will currently throw an ArgNull exception in this case as another data point.